web/src/stream_popover.js

```diff
@@ -562,6 +564,59 @@ export async function build_move_topic_to_stream_popover(
         $("#move_topic_form .move_messages_edit_topic").trigger("focus");
     }
 
+    // The following logic is correct only when
+    // both message_lists.current.data.fetch_status.has_found_newest
+    // and message_lists.current.data.fetch_status.has_found_oldest are true;
+    // otherwise, we cannot be certain of the correct count.
+    function get_count_of_messages_to_be_moved(selected_option) {
+        if (selected_option === "change_one") {
+            return 1;
+        }
+        if (selected_option === "change_later") {
+            return message_util.get_count_of_messages_in_topic_sent_after_current_message(
+                current_stream_id,
+                topic_name,
+                message.id,
+            );
+        }
+        return message_util.get_messages_in_topic(current_stream_id, topic_name).length;
+    }
+
+    function update_move_messages_count_text(selected_option) {
+        const message_move_count = get_count_of_messages_to_be_moved(selected_option);
+        const has_found_newest = message_lists.current.data.fetch_status.has_found_newest();
+        const has_found_oldest = message_lists.current.data.fetch_status.has_found_oldest();
+        const is_topic_narrowed = narrow_state.narrowed_by_topic_reply();
+        const is_stream_narrowed = narrow_state.narrowed_by_stream_reply();
```

Comment:
I think we would also need to check whether the view is narrowed to the same stream or topic for which message is being moved as we can also move topic from the left sidebar popover.

Reply:
Hey @sahil839, I want to confirm one thing. I need to add a check to verify if the messages are being moved to the same stream. If yes, then I should check whether the messages are being moved to the same topic. If both conditions are true, the count should display 0. In all other cases, it should display the count. Am I right?